packages/react/src/components/InAppMessaging/Backdrop/Backdrop.tsx

```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+import { Flex } from '../../../primitives/Flex';
+
+import { BackdropProps } from './types';
+
+export function Backdrop({ children, ...rest }: BackdropProps): JSX.Element {
+  return (
+    <Flex className="amplify-inappmessaging-backdrop" {...rest}>
```

Comment:
Just some general questions:

What are the expected children of Backdrop? Just wondering what is going to be flexed.

Additionally, I'm not sure the exact usage of Backdrop (besides creating the overlay background) so these questions might be more clear when it's put together and ready to review:
- Are the rest of the InAppMessaging components (essentially the modal content) going to be nested inside Backdrop?
- Is there any click-off functionality (click outside to close) the InAppMessage? I'm asking this because a common pattern I've seen is to add a click functionality to the backdrop element, in which case you probably wouldn't want it to have children. i.e. the markup would generally be:
```
<div class="modal-wrapper">
  <div class="backdrop"></div>
  <div class="modal">here is all the modal content with the white background and buttons, and probably modal-wrapper has some flex properties to position me</div>
</div>
```
That would set this up to be re-usable in the future for more general modal functionality, but not necessarily required for this component.

Reply:
Also, if this is the element that wraps the entire component that is overlayed onto the UI (and the rest of the component is nested inside) there are some accessibility requirements missing. Essentially we are mimicking a dialog pattern https://www.w3.org/WAI/ARIA/apg/example-index/dialog-modal/dialog.html Happy to review the implementation details of that!

Reply:
> What are the expected children of Backdrop? Just wondering what is going to be flexed.

> Are the rest of the InAppMessaging components (essentially the modal content) going to be nested inside Backdrop?

The expected children of the Backdrop is essentially modal content. Currently, the plan for executing on a messaging component spec which
* has a backdrop when screen size is non-mobile
* fills the screen when screen size is mobile
```tsx
// non-mobile
<Backdrop>
  <Message />
</Backdrop>


// mobile
<Message />
```

> Is there any click-off functionality (click outside to close) the InAppMessage? I'm asking this because a common pattern I've seen is to add a click functionality to the backdrop element, in which case you probably wouldn't want it to have children.

For this feature, there won't be click-off functionality added to the backdrop. This was something I considered, but the I think the specs call for explicitly using the close button to close the message. For now, the backdrop is sort of considered part of the message - but you raise a good point that this would probably not be as useful for a standalone Backdrop component.

I *think* that, given the fact that we are now narrowly scoping this component to the In-App Messaging feature, it is convenient that the Backdrop also acts as a container  so the message component itself can just try to fill its available space but I am open to changing this. @calebpollman thoughts?

> Also, if this is the element that wraps the entire component that is overlayed onto the UI (and the rest of the component is nested inside) there are some accessibility requirements missing.

Thanks for this callout - it seems like much of what is missing applies to the dialog itself which would be the content. Would you suggest that the backdrop automatically applies some of these missing `aria` properties to its children?